build_config/lib/src/key_normalization.dart

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+String normalizeBuilderKey(String builderKey, String packageName) {
+  if (builderKey.startsWith('|')) return '$packageName$builderKey';
```

Comment:
might as well break this logic out into a function that takes a separator?

Reply:
Done